### PR TITLE
Llama3.2 3B eval

### PR DESCRIPTION
**Pull request by ReemaAlzaid**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (Clean Up)

Please link to any issues this PR addresses: https://github.com/pytorch/torchtune/issues/1810

#### Changelog
What are the changes made in this PR? 

- Copied **evaluation.yaml** to `/recipes/config/llama3_2/` directory
- Updated **evaluation.yam**l to point to Llama-3.2-3B model instantiations
- Updated the recipe registry class to pick up the new Llama-3.2 config

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [x] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

**Llama3.2 3B evaluation recipe output:**
```
(torchtune_env) ➜  torchtune git:(reemaalzaid-move-llama3_2-eval-generation) ✗ tune run eleuther_eval --config llama3_2/evaluation
W1220 03:54:44.815000 59129 torch/distributed/elastic/multiprocessing/redirects.py:29] NOTE: Redirects are currently not supported in Windows or MacOs.
Running EleutherEvalRecipe with resolved config:

batch_size: 8
checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Llama-3.2-3B-Instruct
  checkpoint_files:
  - model-00001-of-00002.safetensors
  - model-00002-of-00002.safetensors
  model_type: LLAMA3_2
  output_dir: ./
  recipe_checkpoint: null
device: cpu
dtype: bf16
enable_kv_cache: true
limit: null
max_seq_length: 4096
model:
  _component_: torchtune.models.llama3_2.llama3_2_3b
quantizer: null
resume_from_checkpoint: false
seed: 1234
tasks:
- truthfulqa_mc2
tokenizer:
  _component_: torchtune.models.llama3.llama3_tokenizer
  max_seq_len: null
  path: /tmp/Llama-3.2-3B-Instruct/original/tokenizer.model

2024-12-20:03:54:45,419 INFO     [_utils.py:28] Running EleutherEvalRecipe with resolved config:

batch_size: 8
checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Llama-3.2-3B-Instruct
  checkpoint_files:
  - model-00001-of-00002.safetensors
  - model-00002-of-00002.safetensors
  model_type: LLAMA3_2
  output_dir: ./
  recipe_checkpoint: null
device: cpu
dtype: bf16
enable_kv_cache: true
limit: null
max_seq_length: 4096
model:
  _component_: torchtune.models.llama3_2.llama3_2_3b
quantizer: null
resume_from_checkpoint: false
seed: 1234
tasks:
- truthfulqa_mc2
tokenizer:
  _component_: torchtune.models.llama3.llama3_tokenizer
  max_seq_len: null
  path: /tmp/Llama-3.2-3B-Instruct/original/tokenizer.model

Model is initialized with precision torch.bfloat16.
2024-12-20:03:55:01,506 INFO     [eleuther_eval.py:503] Model is initialized with precision torch.bfloat16.
2024-12-20:03:55:01,630 INFO     [huggingface.py:132] Using device 'cpu'
config.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 665/665 [00:00<00:00, 470kB/s]
tokenizer_config.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 26.0/26.0 [00:00<00:00, 379kB/s]
vocab.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.04M/1.04M [00:00<00:00, 2.11MB/s]
merges.txt: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 456k/456k [00:00<00:00, 1.42MB/s]
tokenizer.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.36M/1.36M [00:00<00:00, 2.13MB/s]
2024-12-20:03:55:06,104 INFO     [huggingface.py:369] Model parallel was set to False, max memory was not set, and device map was set to {'': 'cpu'}
model.safetensors: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉| 548M/548M [00:42<00:00, 13.0MB/s]
generation_config.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 124/124 [00:00<00:00, 373kB/s]
README.md: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9.59k/9.59k [00:00<00:00, 12.5MB/s]
validation-00000-of-00001.parquet: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 271k/271k [00:00<00:00, 4.88MB/s]
Generating validation split: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 817/817 [00:00<00:00, 29325.28 examples/s]
Running evaluation on the following tasks: ['truthfulqa_mc2']
2024-12-20:03:56:05,340 INFO     [eleuther_eval.py:540] Running evaluation on the following tasks: ['truthfulqa_mc2']
2024-12-20:03:56:05,340 INFO     [task.py:415] Building contexts for truthfulqa_mc2 on rank 0...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 817/817 [00:00<00:00, 2213.25it/s]
2024-12-20:03:56:05,732 INFO     [evaluator.py:496] Running loglikelihood requests
Running loglikelihood requests: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5882/5882 [16:14:46<00:00,  9.94s/it]
Eval completed in 58488.30 seconds.
2024-12-20:20:10:53,643 INFO     [eleuther_eval.py:549] Eval completed in 58488.30 seconds.
Max memory allocated: 0.00 GB
2024-12-20:20:10:53,647 INFO     [eleuther_eval.py:550] Max memory allocated: 0.00 GB


|    Tasks     |Version|Filter|n-shot|Metric|   |Value |   |Stderr|
|--------------|------:|------|-----:|------|---|-----:|---|-----:|
|truthfulqa_mc2|      2|none  |     0|acc   |↑  |0.4975|±  |0.0148|


2024-12-20:20:10:53,933 INFO     [eleuther_eval.py:554] 

|    Tasks     |Version|Filter|n-shot|Metric|   |Value |   |Stderr|
|--------------|------:|------|-----:|------|---|-----:|---|-----:|
|truthfulqa_mc2|      2|none  |     0|acc   |↑  |0.4975|±  |0.0148|
```

### CODE REVIEW

**Review by joecummings — COMMENTED** [review]
This looks great, thanks for the PR! Can you paste some output from running this command to ensure it works properly?

**Review by joecummings — APPROVED** [review]
Looks great - thank you!
